src/v/datalake/record_schema_resolver.cc

```diff
@@ -96,7 +96,7 @@ struct schema_translating_visitor {
             auto type = type_to_iceberg(*d).value();
             co_return type_and_buf{
               .type = resolved_type{
-                .schema = *d,
+                .schema = wrapped_protobuf_descriptor { *d, std::move(pb_def) },
```

Comment:
Good call, right now its indirection. I.e, the descriptor pool is held by `pb_def` in a `shared_ptr`. It may be better to move `pb_def` first then get the reference though in case that changes in the future.

Reply:
ahh i see its heap allocated. got it thanks